Services/Masterdata/Services.Masterdata.EF/Repositories/MasterdataRepository.cs

```diff
@@ -65,6 +66,34 @@ public async Task<ResultPagedListModel<MasterdataModel>> GetMasterdata(string? m
 
         return result;
     }
+
+    public async Task<ResultPagedListModel<string>> GetTags(string masterdataType, QueryModel? querymodel = null)
+    {
+        var pagedResult = await DbContext.Masterdatas
+            .GetByQueryModel(querymodel ?? QueryModel.Default, MasterdataFilter(masterdataType))
+            .ApplyPaging(querymodel ?? QueryModel.Default);
```

Comment:
This method creates a lot of IO overhead. What if you only (distinct) select the tag column (and other columns needed for the MasterDataTagModel) where tag column != null.

You probably need to switch to linq query expresions instead of fluent 
example: 
from x in DbContext.Masterdatas
where ....
select new MasterdataTagModel {
  id = x.id
  ...
}

Reply:
> This method creates a lot of IO overhead. What if you only (distinct) select the tag column (and other columns needed for the MasterDataTagModel) where tag column != null.
> 
> You probably need to switch to linq query expresions instead of fluent example: from x in DbContext.Masterdatas where .... select new MasterdataTagModel { id = x.id ... }

I did this way as I'm not very familiar with linq and doing queries like this, so in this case I may need help!
Also I'm not sure that a select distinct will get me each unique tag (the Tag column of each row, contains an array of tags) from all the filtered rows!

Reply:
I created a performant more simple way of doing this. Check it out, and if fine, please resolve :)

Reply:
> I created a performant more simple way of doing this. Check it out, and if fine, please resolve :)

really nice solution!